File: 1-board-corner-tracking-python/pointCluster.py

class PointCluster:

    def __init__(self):
        self.centroid = (-1, -1)
        self.points = []

    def getCentroid(self):
        return self.centroid

    def add(self, point):
        self.points.append(point)
        self.__updateCentroid()

    def __updateCentroid(self):
        accumulatedY = 0
        accumulatedX = 0
        for point in self.points:
            accumulatedY = accumulatedY + point[0]
            accumulatedX = accumulatedX + point[1]
        numberOfPoints = len(self.points)
        self.centroid[0] = accumulatedY / numberOfPoints
        self.centroid[1] = accumulatedX / numberOfPoints

    def distanceTo(self, point):
        if self.centroid[0] == -1:
            return 0
        return (self.centroid[0] - point[0]) * (self.centroid[0] - point[0]) + (self.centroid[1] - point[1]) * (self.centroid[1] - point[1])

File: 7-improve-corner-candidates-detection/src/PointCluster.java
package src;

import java.util.ArrayList;
import java.util.List;

public class PointCluster {

    private Ponto centroid;
    private List<Ponto> points;

    public PointCluster() {
        centroid = new Ponto(-1, -1);
        points = new ArrayList<>();
    }

    public Ponto getCentroid() { return centroid; }

    public void add(Ponto point) {
        points.add(point);
        updateCentroid();
    }

    private void updateCentroid() {
        int accumulatedY = 0;
        int accumulatedX = 0;
        for (Ponto point: points) {
            accumulatedY = accumulatedY + point.y;
            accumulatedX = accumulatedX + point.x;
        }
        centroid.y = accumulatedY / points.size();
        centroid.x = accumulatedX / points.size();
    }

    public double distanceTo(Ponto point) {
        if (centroid.y == -1) return 0;
        return centroid.distanceTo(point);
    }

}

